.github/workflows/test-linux.yml

```
@@ -31,3 +31,6 @@ jobs:
         run: make build-ci -j2 V=1 CONFIG_FLAGS="--error-on-warn"
       - name: Test
         run: make run-ci -j2 V=1 TEST_CI_ARGS="-p actions"
+      - name: Test Internet
+        # Only run if files in test/internet have changed.
+        run: git diff --name-only origin/HEAD | grep test/internet && make test-internet -j2 V=1
```

Maybe we should at least also specify the path in the 'git diff' command so that we know it will only check for changes in that directory, in case the grep might accidentally match something else in a dependency for example (e.g. npm or its dependencies).
```suggestion
        run: git diff --name-only origin/HEAD -- test/internet | grep test/internet && make test-internet -j2 V=1
```

If we wanted to avoid duplicating the name we could check if `wc -l` output > 1 or maybe just check if we had any non-empty lines:

```suggestion
        run: git diff --name-only origin/HEAD -- test/internet | grep -v '^$' && make test-internet -j2 V=1
```

---

Using some of the suggestion from @mscedex but also dropping `grep` because it returns an error code if it finds nothing, and I suspect whatever I have to do to swallow its errors but not the `make` command's errors will be less readable than using an `if` in the shell:

```suggestion
        run: if [[ $(git diff --name-only origin/HEAD -- test/internet) ]]; then make test-internet -j2 V=1; fi
```